packages/aws-cdk/lib/api/hotswap/common.ts

```diff
@@ -1,24 +1,35 @@
-import type { PropertyDifference, Resource } from '@aws-cdk/cloudformation-diff';
+import type { PropertyDifference } from '@aws-cdk/cloudformation-diff';
+import type { HotswappableChange, ResourceChange } from '../../../../@aws-cdk/tmp-toolkit-helpers/src/api/io/payloads/hotswap';
 import { ToolkitError } from '../../toolkit/error';
 import type { SDK } from '../aws-auth';
 
 export const ICON = '✨';
 
-export interface HotswappableChange {
+export interface HotswapOperation {
+  /**
+   * Marks the operation as hotswappable
+   */
   readonly hotswappable: true;
-  readonly resourceType: string;
-  readonly propsChanged: Array<string>;
```

Comment:
Surprisingly, these were unused. `propsChanged` is not made available as raw data via `change`.